log4j-api/src/main/java/org/apache/logging/log4j/message/ParameterFormatter.java

```diff
@@ -240,12 +244,14 @@ static void formatMessage(
             return;
         }
 
-        // Fail if there are insufficient arguments
-        if (analysis.placeholderCount > args.length) {
+        // check if there are insufficient arguments that do not include Throwable arg
+        final int realArgCount = args.length;
+        final int noThrowableArgCount = realArgCount - ((args[realArgCount - 1] instanceof Throwable) ? 1 : 0);
+        if (analysis.placeholderCount != noThrowableArgCount) {
```

Comment:
Ah, you mean that code like:

```java
logger.info("An error in {} occurred: {}", "foo", new RuntimeException());
```

should give a warning? I might agree that it should give a warning, since such statements prevent users from switching to Logback in the future. However for backward compatibility, such statements should be silently accepted.

@vy, what do you think?

Reply:
Please confirm if I need to give a warning log. I will change my codes or not. @vy @ppkarwasz